nautobot/core/management/commands/pre_migrate.py

```diff
@@ -0,0 +1,149 @@
+            f"{list(interfaces_with_multiple_vrfs)}\n"
+            f"{list(interfaces_with_mixed_vrfs)}\n"
+            f"{list(vm_interfaces_with_multiple_vrfs)}\n"
+            f"{list(vm_interfaces_with_mixed_vrfs)}"
+        )
+
+
+def check_virtualchassis_uniqueness():
+    """
+    Check for uniqueness enforcement changes for VirtualChassis.
+
+    - Make `name` unique, reject migration if duplicate (don't want to rename VCs)
+
+    See: https://github.com/nautobot/nautobot/issues/3846
+    """
+    vc_dupes = VirtualChassis.objects.values("name").annotate(count=models.Count("id")).filter(count__gt=1)
+
+    if vc_dupes.exists():
+        raise ValidationError(
+            "You cannot migrate VirtualChassis objects that non-unique names:\n" f"{list(vc_dupes)}\n"
+        )
+
+
+def check_exporttemplate_uniqueness():
+    """
+    Check for uniqueness enforcement changes for ExportTemplate.
+
+    - Move to `unique_together` on just `content_type` and `name`, reject migration if duplicate.
+
+    See: https://github.com/nautobot/nautobot/issues/3848
+    """
+    et_dupes = (
+        ExportTemplate.objects.values("content_type", "name").annotate(count=models.Count("id")).filter(count__gt=1)
+    )
+
+    if et_dupes.exists():
+        raise ValidationError(
+            f"You cannot migrate ExportTemplate objects with non-unique content_type, name pairs:\n {list(et_dupes)}\n"
+        )
+
+
+def check_configcontext_uniqueness():
+    """
+    Check for uniqueness enforcement changes for ConfigContext and ConfigContextSchema.
+
+    - Move to `name` unique, reject migration if duplicate
+
+    See: https://github.com/nautobot/nautobot/issues/3849
+    """
+    cc_dupes = ConfigContext.objects.values("name").annotate(count=models.Count("id")).filter(count__gt=1)
+    ccs_dupes = ConfigContextSchema.objects.values("name").annotate(count=models.Count("id")).filter(count__gt=1)
+
+    if any(
+        [
+            cc_dupes.exists(),
+            ccs_dupes.exists(),
+        ]
+    ):
+        raise ValidationError(
+            "You cannot migrate ConfigContext or ConfigContextSchema objects that have non-unique names:\n"
+            f"{list(cc_dupes)}\n"
+            f"{list(ccs_dupes)}\n"
+        )
+
+
+class Command(BaseCommand):
+    help = HELP_TEXT
+
+    def create_parser(self, *args, **kwargs):
+        """Custom parser that can display multiline help."""
+        parser = super().create_parser(*args, **kwargs)
+        parser.formatter_class = argparse.RawTextHelpFormatter
+        return parser
+
+    def handle(self, *args, **options):
+        # Pre-migration checks run here.
+
+        checks = [
+            check_interface_vrfs,
+            check_configcontext_uniqueness,
+            check_exporttemplate_uniqueness,
+            check_virtualchassis_uniqueness,
+        ]
+
+        for check, args in checks:
+            try:
+                print(f">>> Running check: {check}...")
```

Comment:
Management commands are [recommended](https://docs.djangoproject.com/en/4.2/howto/custom-management-commands/) to use `self.stdout.write` rather than `print`